google/cloud/spanner/benchmarks/multiple_rows_cpu_benchmark.cc

```diff
@@ -395,13 +381,147 @@ class ReadExperiment : public Experiment {
+  };
+
+  void DumpSamples(std::vector<RowCpuSample> const& samples) const {
+    std::lock_guard<std::mutex> lk(mu_);
+    std::copy(samples.begin(), samples.end(),
+              std::ostream_iterator<RowCpuSample>(std::cout, "\n"));
+    auto it =
+        std::find_if(samples.begin(), samples.end(),
+                     [](RowCpuSample const& x) { return !x.status.ok(); });
+    if (it == samples.end()) {
+      return;
+    }
+    std::cout << "# FIRST ERROR: " << it->status << "\n";
+  }
+
+  void LogError(std::string const& s) {
+    std::lock_guard<std::mutex> lk(mu_);
+    std::cout << "# " << s << std::endl;
+  }
+
+ private:
+  Status FillTableTask(Config const& config, cs::Client client,
+                       std::string const& table_name, int task_count,
+                       int task_id) {
+    std::vector<std::string> const column_names{
+        "Key",   "Data0", "Data1", "Data2", "Data3", "Data4",
+        "Data5", "Data6", "Data7", "Data8", "Data9"};
+    using T = typename Traits::native_type;
+    T value0 = GenerateRandomValue();
+    T value1 = GenerateRandomValue();
+    T value2 = GenerateRandomValue();
+    T value3 = GenerateRandomValue();
+    T value4 = GenerateRandomValue();
+    T value5 = GenerateRandomValue();
+    T value6 = GenerateRandomValue();
+    T value7 = GenerateRandomValue();
+    T value8 = GenerateRandomValue();
+    T value9 = GenerateRandomValue();
+
+    auto mutation = cs::InsertOrUpdateMutationBuilder(table_name, column_names);
+    int current_mutations = 0;
+
+    auto maybe_flush = [&, this](bool force) -> Status {
+      if (current_mutations == 0) {
+        return {};
+      }
+      if (!force && current_mutations < 1000) {
+        return {};
+      }
+      auto m = std::move(mutation).Build();
+      auto result = client.Commit(
+          [&m](cs::Transaction const&) { return cs::Mutations{m}; });
+      if (!result) {
+        std::lock_guard<std::mutex> lk(mu_);
+        std::cerr << "# Error in Commit() " << result.status() << "\n";
+        return std::move(result).status();
+      }
+      mutation = cs::InsertOrUpdateMutationBuilder(table_name, column_names);
+      current_mutations = 0;
+      return {};
+    };
+    auto force_flush = [&maybe_flush] { return maybe_flush(true); };
+    auto flush_as_needed = [&maybe_flush] { return maybe_flush(false); };
+
+    auto const report_period =
+        (std::max)(static_cast<std::int64_t>(2), config.table_size / 50);
+    for (std::int64_t key = 0; key != config.table_size; ++key) {
+      // Each thread does a fraction of the key space.
+      if (key % task_count != task_id) continue;
+      // Have one of the threads report progress about 50 times.
+      if (task_id == 0 && key % report_period == 0) {
+        std::cout << '.' << std::flush;
+      }
+      mutation.EmplaceRow(key, value0, value1, value2, value3, value4, value5,
+                          value6, value7, value8, value9);
+      current_mutations++;
```

Comment:
Nit: Using post-increment when you don't need post-increment semantics will cause some human readers to skim back in the code to find the type of the variable being post-incremented to see if the use of post-increment is a pessimization or equivalent to pre-increment. As a general rule of thumb, I'd say use post-increment IFF you need post-increment semantics. Otherwise, use pre-increment.

Reply:
Done and rebased.